### PR TITLE
Added thread.currentid()

### PR DESCRIPTION
**Pull request by NiLSPACE**
Makes it easier to debug the Lua code to identify which thread is running

### CODE REVIEW

**Review by madmaxoft — CHANGES_REQUESTED** [review]
I think the code pushing the thread ID onto Lua stack should be hoisted to a separate function.

**Comment by madmaxoft** [comment]
Also, ideally you should add a script into /tests that tests / shows the functionality.

File: src/Utils.h [hunk 1]
```diff
@@ -23,3 +25,6 @@ std::string lua_getstring(lua_State * L, int aIndex);
 
 /** Returns the value of the named field of a table at the specified index, as a std::string. */
 std::string lua_getstringfield(lua_State * L, int aTableIndex, const char * aFieldName);
+
+/** Pushes the value of a std::thread::id into a lua stack as a string */
+int pushThreadIdOnLuaStack(lua_State* aState, std::thread::id aThreadId);
```
Comment (madmaxoft) [inline src/Utils.h]:
```suggestion
int pushThreadIdOnLuaStack(lua_State * aState, const std::thread::id & aThreadId);
```

File: tests/CurrentThreadId.lua [hunk 2]
```diff
@@ -0,0 +1,21 @@
+-- Tests the thread.currentid() function.
+
+local thread = require("thread");
+
+-- Variable which the sub thread will save it's id in.
+local threadId;
+
+-- Create a new thread which will save it's id to the threadId variable after one second.
+-- The extra second is to give the main thread more time to retrieve the id.
```
Comment (madmaxoft) [inline tests/CurrentThreadId.lua]: One second may be too much, `0.1` should be enough.

We'll get rid of this anyway, when we add mutexes / condvars / semaphores

File: tests/CurrentThreadId.lua [hunk 3]
```diff
@@ -0,0 +1,21 @@
+-- Tests the thread.currentid() function.
+
+local thread = require("thread");
+
+-- Variable which the sub thread will save it's id in.
```
Comment (madmaxoft) [inline tests/CurrentThreadId.lua]:
```suggestion
-- Variable which the sub thread will save its id in.
```

**Review by madmaxoft — CHANGES_REQUESTED** [review]

**Comment by NiLSPACE** [comment]
That should do it :)

**Review by madmaxoft — APPROVED** [review]

**Pull request merged**